src/makerom/entry.s

```diff
@@ -1,34 +1,45 @@
 #include "ultra64/asm.h"
+#include "ultra64/regdef.h"
 #include "boot.h"
 
-.set noreorder
+.text
 
-.section .text
-
-.balign 16
-
-LEAF(entrypoint)
-    // Clear boot segment .bss
-    la      $t0, _bootSegmentBssStart
-#ifndef AVOID_UB
-    // UB: li only loads the lower 16 bits of _bootSegmentBssSize when it may be larger than this,
-    // so not all of bss may be cleared if it is too large
-    li      $t1, _bootSegmentBssSize
+#if defined(__sgi) && !defined(AVOID_UB)
+/* IDO assembler workaround: The makerom tool in the N64 SDK was given the bss segment size as a const
+ * literal, and since this literal was < 0x10000 it was loaded in one instruction. We don't have access
+ * to the bss segment size until we link everything so we cannot do the same thing. We can load the
+ * lower 16 bits of the bss size; however IDO emits two R_MIPS_LO16 relocations at this instruction,
+ * so we must use a special symbol that is the bss size divided by two so that the linker updates the
+ * value twice to the correct bss size.
+ * When AVOID_UB is enabled, don't do this and instead load the full symbol value. */
+#define LOAD_BSS_SIZE(reg) addiu reg, zero, %lo(_bootSegmentBssSize_2)
```

Comment:
lol, wtf. Is there anything else we can do about this? Maybe we should keep the entire makerom portion as GCC, since the rom header already is and there's more weirdness with the padding below

Reply:
I specifically wanted to move entry.s to IDO to reduce ugly diffs in iQue, imo either way looks awful but at least this way it can be tucked away nicer

Reply:
> at least this way it can be tucked away nicer
well idk about that, it took me a while to understand this.

How about
```
#define LOAD_BSS_SIZE(reg) addiu reg, zero, %half(_bootSegmentBssSize)
```
instead?

Reply:
I tried it and it seems to work, though not sure if there's unintended consequences

Reply:
Seems good to me